### PR TITLE
[fix](resource)fix check available fail when s3 aws_token  is set and reset as, sk faild on be.

### PR DESCRIPTION
**Pull request by kobe6th**
## Proposed changes

Issue Number: close #xxx

1. fe return 'S3 can't use, please check your properties' when AWS_TOKEN is set
`
CREATE RESOURCE "test_resources"
PROPERTIES
(
    "type" = "s3",
    "AWS_ENDPOINT" = "http://cos.ap-guangzhou.myqcloud.com",
    "AWS_ACCESS_KEY"="ak",
    "AWS_SECRET_KEY"="sk",
    "AWS_TOKEN"="token",
    "AWS_REGION" = "ap-guangzhou",
    "AWS_BUCKET" = "bucket",
    "AWS_ROOT_PATH" = "path/to/dir",    
    "AWS_MAX_CONNECTIONS" = "50",
    "AWS_REQUEST_TIMEOUT_MS" = "3000",
    "AWS_CONNECTION_TIMEOUT_MS" = "1000"
);

ERROR 1105 (HY000): errCode = 2, detailMessage = S3 can't use, please check your properties
`

2. When altering the AK/SK resource, only the AK and SK are set in the S3ClientHolder::reset method. Therefore, creating a new S3 client without other parameters is not possible.

### CODE REVIEW

**Comment by nextdreamblue** [comment]
run buildall

**Comment by nextdreamblue** [comment]
run buildall

File: be/src/io/fs/s3_file_system.cpp [hunk 1]
```diff
@@ -134,8 +139,7 @@ Status S3ClientHolder::reset(const S3ClientConf& conf) {
     {
         std::lock_guard lock(_mtx);
         _client = std::move(client);
-        _conf.ak = conf.ak;
-        _conf.sk = conf.sk;
+        _conf = std::move(reset_conf);
```
Comment (platoneko) [inline be/src/io/fs/s3_file_system.cpp:142]: 这样改和在这儿去 `_conf.token = conf.token`  不是等价的吗？

Comment (kobe6th) [reply be/src/io/fs/s3_file_system.cpp:142]: 是等价的，目前这样写可以少2行代码。

**Review by platoneko — APPROVED** [review]

**Review by lide-reed — APPROVED** [review]
LGTM

**Pull request merged**